boilerplate/_lib/ensure.sh

```diff
@@ -3,17 +3,27 @@ set -euo pipefail
 
 GOLANGCI_LINT_VERSION="1.30.0"
 DEPENDENCY=${1:-}
+GOOS=$(go env GOOS)
 
 case "${DEPENDENCY}" in
-    golangci-lint)
-        GOPATH=$(go env GOPATH)
-        if [ ! -f "${GOPATH}/bin/golangci-lint" ]; then
-            DOWNLOAD_URL="https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz"
-            curl -sfL "${DOWNLOAD_URL}" | tar -C "${GOPATH}/bin" -zx --strip-components=1 "golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64/golangci-lint"
+golangci-lint)
+    GOPATH=$(go env GOPATH)
+    if which golangci-lint ; then
+        exit
+    else
+        mkdir -p "${GOPATH}/bin"
+        echo "${PATH}" | grep -q "${GOPATH}/bin"
+        IN_PATH=$?
+        if [ $IN_PATH != 0 ]; then
+            echo "${GOPATH}/bin not in $$PATH"
+            exit 1
         fi
+        DOWNLOAD_URL="https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-${GOOS}-amd64.tar.gz"
+        curl -sfL "${DOWNLOAD_URL}" | tar -C "${GOPATH}/bin" -zx --strip-components=1 "golangci-lint-${GOLANGCI_LINT_VERSION}-${GOOS}-amd64/golangci-lint"
```

Comment:
Doesn't the `amd64` bit also break on Mac? (`ppc` or something?)

Reply:
No Macs have used intel x86 architecture for about 10 years now.

Reply:
TIL.

We're still totally screwing all the people trying to develop on AIX though.